src/App.tsx

```diff
@@ -1,223 +1,76 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+/* eslint-disable curly */
+
+import React, { useContext, useEffect, useState } from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { getTodos } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [isError, setIsError] = useState<Error | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+  const getTodosList = () => {
+    if (!user) {
+      return;
     }
+
+    getTodos(user.id)
+      .then(data => setTodos(data))
+      .catch(() => setIsError(Error.Update));
+  };
+
+  const filterdTodos = () => {
+    if (!todos) return null;
+
+    return todos?.filter((todo) => {
+      switch (filter) {
+        case Filter.Active: return !todo.completed;
+        case Filter.Completed: return todo.completed;
+        case Filter.All:
+        default:
+          return true;
+      }
+    });
+  };
+
+  const getFilterStatus = (arg: Filter) => {
+    setFilter(arg);
+  };
```

Comment:
if your function name starts with "get" - it is expected to return something that we are trying to get, so use anothe name